Signal/src/Models/SyncPushTokensJob.swift

```diff
@@ -27,41 +27,55 @@ class SyncPushTokensJob: NSObject {
     func run() -> Promise<Void> {
         Logger.info("\(TAG) Starting.")
 
-        // Required to potentially prompt user for notifications settings
-        // before `requestPushTokens` will return.
-        self.pushManager.validateUserNotificationSettings()
-
-        let runPromise: Promise<Void> = self.requestPushTokens().then { (pushToken: String, voipToken: String) in
-            var shouldUploadTokens = false
-
-            if self.preferences.getPushToken() != pushToken || self.preferences.getVoipToken() != voipToken {
-                Logger.debug("\(self.TAG) Push tokens changed.")
-                shouldUploadTokens = true
-            } else if !self.uploadOnlyIfStale {
-                Logger.debug("\(self.TAG) Uploading even though tokens didn't change.")
-                shouldUploadTokens = true
-            }
-
-            Logger.warn("\(self.TAG) lastAppVersion: \(AppVersion.instance().lastAppVersion), currentAppVersion: \(AppVersion.instance().currentAppVersion)")
-            if AppVersion.instance().lastAppVersion != AppVersion.instance().currentAppVersion {
-                Logger.debug("\(self.TAG) Fresh install or app upgrade.")
-                shouldUploadTokens = true
-            }
-
-            guard shouldUploadTokens else {
-                Logger.warn("\(self.TAG) Skipping push token upload. pushToken: \(pushToken), voipToken: \(voipToken)")
-                return Promise(value: ())
-            }
-
-            Logger.warn("\(self.TAG) Sending new tokens to account servers. pushToken: \(pushToken), voipToken: \(voipToken)")
-
-            return self.accountManager.updatePushTokens(pushToken:pushToken, voipToken:voipToken).then {
-                return self.recordNewPushTokens(pushToken:pushToken, voipToken:voipToken)
-            }.then {
-                Logger.debug("\(self.TAG) Successfully ran syncPushTokensJob.")
-            }.catch { error in
-                Logger.error("\(self.TAG) Failed to run syncPushTokensJob with error: \(error).")
-            }
+        let runPromise: Promise<Void> =
+            Promise { fulfill, _ in
+                // TODO: PromiseKit only lets you specify the queue in `then`,
+                //       so use a noop promise here so that we can specify the
+                //       queues to use below.
+                fulfill()
```

Comment:
Doesn't PromiseKit have a simpler way to create a new promise that is immediately fulfilled?

Reply:
Done, thanks.